crates/sqlexec/src/extension_codec.rs

```diff
@@ -991,3 +1141,268 @@ impl<'a> PhysicalExtensionCodec for GlareDBExtensionCodec<'a> {
+            name,
+        },
+    )
+}
+
+fn aggr_expr_to_window_fn(
+    expr: &dyn AggregateExpr,
+) -> Result<physical_window_expr_node::WindowFunction> {
+    let aggr_expr = expr.as_any();
+    // TODO: `DistinctCount`, `DistinctSum`, `DistinctBitXor`,
+    // `DistinctArrayAgg`, `OrderSensitiveArrayAgg` ?
+    let func = if aggr_expr.downcast_ref::<Count>().is_some() {
+        datafusion_proto::protobuf::AggregateFunction::Count
+    } else if aggr_expr.downcast_ref::<Grouping>().is_some() {
+        datafusion_proto::protobuf::AggregateFunction::Grouping
+    } else if aggr_expr.downcast_ref::<BitAnd>().is_some() {
+        datafusion_proto::protobuf::AggregateFunction::BitAnd
+    } else if aggr_expr.downcast_ref::<BitOr>().is_some() {
+        datafusion_proto::protobuf::AggregateFunction::BitOr
+    } else if aggr_expr.downcast_ref::<BitXor>().is_some() {
+        datafusion_proto::protobuf::AggregateFunction::BitXor
+    } else if aggr_expr.downcast_ref::<BoolAnd>().is_some() {
+        datafusion_proto::protobuf::AggregateFunction::BoolAnd
+    } else if aggr_expr.downcast_ref::<BoolOr>().is_some() {
+        datafusion_proto::protobuf::AggregateFunction::BoolOr
+    } else if aggr_expr.downcast_ref::<Sum>().is_some() {
+        datafusion_proto::protobuf::AggregateFunction::Sum
+    } else if aggr_expr.downcast_ref::<ApproxDistinct>().is_some() {
+        datafusion_proto::protobuf::AggregateFunction::ApproxDistinct
+    } else if aggr_expr.downcast_ref::<ArrayAgg>().is_some() {
+        datafusion_proto::protobuf::AggregateFunction::ArrayAgg
+    } else if aggr_expr.downcast_ref::<Min>().is_some() {
+        datafusion_proto::protobuf::AggregateFunction::Min
+    } else if aggr_expr.downcast_ref::<Max>().is_some() {
+        datafusion_proto::protobuf::AggregateFunction::Max
+    } else if aggr_expr.downcast_ref::<Avg>().is_some() {
+        datafusion_proto::protobuf::AggregateFunction::Avg
+    } else if aggr_expr.downcast_ref::<Variance>().is_some() {
+        datafusion_proto::protobuf::AggregateFunction::Variance
+    } else if aggr_expr.downcast_ref::<VariancePop>().is_some() {
+        datafusion_proto::protobuf::AggregateFunction::VariancePop
+    } else if aggr_expr.downcast_ref::<Covariance>().is_some() {
+        datafusion_proto::protobuf::AggregateFunction::Covariance
+    } else if aggr_expr.downcast_ref::<CovariancePop>().is_some() {
+        datafusion_proto::protobuf::AggregateFunction::CovariancePop
+    } else if aggr_expr.downcast_ref::<Stddev>().is_some() {
+        datafusion_proto::protobuf::AggregateFunction::Stddev
+    } else if aggr_expr.downcast_ref::<StddevPop>().is_some() {
+        datafusion_proto::protobuf::AggregateFunction::StddevPop
+    } else if aggr_expr.downcast_ref::<Correlation>().is_some() {
+        datafusion_proto::protobuf::AggregateFunction::Correlation
+    } else if let Some(_regr_expr) = aggr_expr.downcast_ref::<Regr>() {
+        // Can't get regr_type without public RegrType
```

Comment:
is there a ticket for this? do we intend to do this before the merge?

Reply:
Yup, everything is in here: https://github.com/apache/arrow-datafusion/pull/7557